### PR TITLE
Skip prepare_serial_console() for non-qemu backends

### PR DESCRIPTION
**Pull request by pevik**
This should fix RPI and jeos-main@svirt-xen-hvm failures reported in https://github.com/os-autoinst/os-autoinst-distri-opensuse/commit/4a89bfb0f44e961b9c5a9b9f5d152391c0bb40d3#r98153902

More notes in: https://progress.opensuse.org/issues/122935#note-13

Verification run:
- https://openqa.suse.de/tests/overview?build=fix/prepare_serial_console
- https://openqa.opensuse.org/tests/overview?build=fix/prepare_serial_console
Tests which were broken:
- RPi3 https://openqa.opensuse.org/tests/3074834#step/system_prepare/3
- RPi4: https://openqa.opensuse.org/tests/3074835#step/system_prepare/3

@ggardet @mloviska FYI
@os-autoinst/tests-maintainer Could you please review this? I suppose I'm correct that starting getty is needed really only on virtio, but maybe I'm wrong.

### CODE REVIEW

**Comment by ggardet** [comment]
This looks OK on RPi3: https://openqa.opensuse.org/tests/3074834#step/system_prepare/3
and RPi4: https://openqa.opensuse.org/tests/3074835#step/system_prepare/3

Thanks!

File: lib/serial_terminal.pm [hunk 1]
```diff
@@ -69,26 +69,37 @@ for ppc64le).
...
+    # *enabled*).
+    unless (!check_var('VIRTIO_CONSOLE', 0)) {
+        record_info('skip virtio', 'Skipped due disabled virtio console');
+        return;
+    }
 
-    if (!check_var('VIRTIO_CONSOLE', 0)) {
-        my $console = 'hvc1';
+    unless (is_qemu) {
+        record_info('skip backend', 'Skipped due unsupported arch (only BACKEND=qemu supported)');
```
Comment (Martchus) [inline lib/serial_terminal.pm:81]:
```suggestion
        record_info('skip backend', 'Skipped due unsupported backend (only BACKEND=qemu supported)');
```

Maybe it makes also sense to state *what* has been skipped (not sure whether it'll be clear in the context the info is recorded):
```suggestion
        record_info('skip virtio', 'Skipped adding consoles due unsupported backend (only BACKEND=qemu supported)');
```

File: lib/serial_terminal.pm [hunk 2]
```diff
@@ -69,26 +69,37 @@ for ppc64le).
 =cut
 
 sub prepare_serial_console {
-    record_info('getty before', script_output('systemctl | grep serial-getty'));
+    # Virtio console is disabled only with VIRTIO_CONSOLE=0 (by default
+    # VIRTIO_CONSOLE is not set which also means that virtio console is
+    # *enabled*).
+    unless (!check_var('VIRTIO_CONSOLE', 0)) {
+        record_info('skip virtio', 'Skipped due disabled virtio console');
```
Comment (Martchus) [inline lib/serial_terminal.pm:76]:
```suggestion
        record_info('skip virtio', 'Skipped due to disabled virtio console');
```

File: lib/serial_terminal.pm (hunk 1 above)
Comment (pevik) [reply lib/serial_terminal.pm:81]: Thank you, both makes sense. I had different title `skip virtio` vs. `skip backend` to see the reason without click. But I'll change it (it's better to click that have misleading title).

**Comment by pevik** [comment]
FYI I plan to merge tonight, so that we have fixed openQA during hackweek.

**Pull request merged**